docs/additional-options.md

```diff
@@ -0,0 +1,105 @@
+  public PersonValidator() {
+        RuleFor(person => person.Surname).NotNull();
+        RuleFor(person => person.Forename).NotNull();
+  }
+}
+```
+
+By default, if these rules fail they will have a severity of "Error". This can be changed by calling the `WithSeverity` method. For example, if we wanted a missing surname to be identified as a warning instead of an error then we could modify the above line to:
+
+```csharp
+RuleFor(x => x.Surname).NotNull().WithSeverity(Severity.Warning);
+```
+
+In version 9.0 and above a callback can be used instead, which also gives you access to the item being validated:
+
+```csharp
+RuleFor(person => person.Surname).NotNull().WithSeverity(person => Severity.Warning);
+```
+
+In this case, the `ValidationResult` would still have an `IsValid` result of `false`. However, in the list of `Errors`, the `ValidationFailure` pertaining to this field will have its `Severity` property set to `Warning`:
+
+```csharp
+var validator = new PersonValidator();
+var result = validator.Validate(new Person());
+foreach (var failure in result.Errors) {
+  Console.WriteLine($"Property: {failure.PropertyName} Severity: {failure.Severity}");
+}
+```
+
+The output would be:
+
+```
+Property: Surname Severity: Warning
+Property: Forename Severity: Error
+```
+
+By default, the severity level of every validation rule is `Error`. Available options are `Error`, `Warning`, or `Info`.
+
+## Returning Custom Codes using ErrorCode
+
+A custom error code can also be associated with validation rules by using the `WithErrorCode` method. We could modify the Surname rule with the following:
+
+```csharp
+RuleFor(person => person.Surname).NotNull().WithErrorCode("ERR1234");
+```
+
+The resulting error code can be obtained from the `ErrorCode` property on the `ValidationFailure`:
+
+```csharp
+var validator = new PersonValidator();
+var result = validator.Validate(new Person());
+foreach (var failure in result.Errors) {
+  Console.WriteLine($"Property: {failure.PropertyName} Error Code: {failure.ErrorCode}");
+}
+```
+
+The output would be:
+
+```
+Property: Surname Error Code: ERR1234
+Property: Forename Error Code: NotNullValidator
+```
+
+Note that if an error code is not explicitly specified, it will default to the name of the underlying property validator (eg for a call to `NotNull` it would be `"NotNullValidator"`).
+
+Note that this affects the localized lookups of error messages as well, as the error code is used for the localization key. If you override the error code in a localized environment, you'll need to ensure that you provide a localized value using that error code as well. When there is a custom error code with no key, FluentValidation falls back to the standard key. For example, if you provide an error code of `ERR123` on a `NotNull` validation, you'll want to have a localization value by `ERR123` in your resource file. If you do not supply a translation, the resource for `NotNullValidator` will be used in its place.
```

Comment:
> you'll need to ensure that you provide a localized value using that error code as well

Maybe I'm misreading this, but that's not the case. If there's no localized lookup using the error code it still uses the default.

Reply:
Ie you can do `.Must(x => ...).WithErrorCode("NotNullValidator")` to inherit the error message for the notnullvalidator

Reply:
I see -- I misunderstood what we were trying to accomplish with that. I'll fix it up. 👍